### DISCUSSION TITLE
Have an option for automatic periodic exports/backups

### DISCUSSION THREAD

**Discussion by shrinidhi111** [outline/outline#5022] [Ideas]
For backups, we can

- Take a backup of all the data files
- Take a backup of the Postgres
- Use the "export" option within Outline itself

Personally, I find the export option most convenient as it retains all the data needed and can be easily exported

It would be great if there was a feature to automatically export data at a periodic cron time. Ideally to a S3 would be even better

**Comment by tommoor** [comment]
I feel like this functionality is easily implemented outside the codebase, as you want something that is not tied directly to the thing it's backing up. There are API methods provided for export – you can set up a script to jump to S3 or storage of your choice.

**Reply by shrinidhi111** [reply]
Thanks for your input @tommoor !


Given the need for backups for admins, I would assume many of them would benefit from an official backup mechanism.


I think maybe a simple section in README with the APIs for backup can be a starting point

**Reply by tommoor** [reply]
Just added some details here: https://docs.getoutline.com/s/hosting/doc/backups-KZtPOADCHG – ultimately in the self-hosted environment I don't think that the API approach is very useful, just backup the entire database.

**Comment by prio** [comment]
If its any use I have a small Python script I run via cron that runs an export and unzips it into a git repo https://gist.github.com/prio/e40b01262ed696c0c8538f1fae56a3b8

The shell script wrapper then pushes the changes to github so I have a searchable/viewable copy on Github if my server goes down.

**Reply by shrinidhi111** [reply]
Thank you @prio !

This helps

**Comment by routerino** [comment]
These days you have many options to back up an application without the application having to be aware. For myself, I take hourly `btrfs` snapshots of the container config and volumes, replicate them, and then encrypt them offsite using [kopia](https://blog.gurucomputing.com.au/s/blog/doc/offsite-docker-backups-with-kopia-pPUqvp0V4Q). No manual intervention necessary.

Actually that's how that blog post is currently being backed up.
